util/chplenv/defaults.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+"""
+Provides simple 'get()' interface for accessing default value overrides
+Checks environment variables first, then chplconfig file for definitions
+"""
+from __future__ import print_function
+import os, sys
+
+chplenv_dir = os.path.dirname(__file__)
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+from utils import chplvars, memoize
+
+
+class ChapelConfig(object):
+    """ Class for parsing chplconfig file and providing 'get' interface """
+    def __init__(self):
+        """ Find and parse chplconfig file, populating self.chplconfig """
+
+        # Dictionary containing all of the env vars defined in chplconfig
+        self.chplconfig = {}
+
+        # List of warnings to track if a warning has occurred
+        self.warnings = []
+
+        # Find chplconfigfile and prettypath
+        self.find()
+
+        # Populate chplconfig
+        if self.chplconfigfile:
+            self.parse()
+
+        # Print any warnings accumulated throughout constructor
+        if self.warnings:
+            self.printwarnings()
+
+    def get(self, var):
+        """ Wrapper for chplconfig[var] intended to mirror os.environ.get() """
+        if var in self.chplconfig.keys():
+            return self.chplconfig[var]
+
+    def find(self):
+        """ Find chplconfig file path"""
+
+        # Places to look for a chplconfig file
+        chplconfigpath = os.environ.get('CHPL_CONFIG')
+        home = os.path.expanduser('~')
```

Comment:
`home` is only used once, maybe just say `ccpath = os.path.expanduser('~')` below instead?